src/components/message/message-card/MessageCardBase.tsx

```diff
@@ -48,168 +49,201 @@ export interface MessageCardBaseProps {
+const MessageCardBase = React.memo(
+	({
+		message,
 		viewType,
-		messageId,
-		messageBody: body,
-		isBeautified: viewType === MessageViewType.FORMATTED,
-		rawContent: bodyBase64,
-		isSelected: isAttached || false,
+		setViewType,
+		hoverMessage,
+		unhoverMessage,
+		isAttached,
+		isBookmarked,
+		isHighlighted,
+		isSoftFiltered,
+		toogleMessagePin,
+		isEmbedded,
 		isDetailed,
+		isExported,
+		isExport,
+		addMessageToExport,
 		applyFilterToBody,
-	};
-
-	const messageCardToolsConfig: MessageCardToolsConfig = {
-		message,
-		messageId,
-		messageType,
-		messageViewType: viewType,
-		toggleViewType,
-		isBookmarked: isBookmarked || false,
-		toggleMessagePin: toogleMessagePin || (() => null),
-		isScreenshotMsg,
-		isEmbedded,
-	};
-
-	const renderMessageInfo = () => {
-		if (viewType === MessageViewType.FORMATTED || viewType === MessageViewType.BINARY) {
-			const formattedTimestamp = formatTime(timestampToNumber(timestamp));
-			return (
-				<div className='mc-header__info'>
-					<div
-						className='mc-header__value mc-header__timestamp'
-						title={`Timestamp: ${formattedTimestamp}`}
-						onMouseEnter={hoverMessage}
-						onMouseLeave={unhoverMessage}>
-						{timestamp && formattedTimestamp}
-					</div>
-					<div className='mc-header__item' title={`Session: ${sessionId}`}>
-						<span className={sessionClass} style={sessionArrowStyle} />
-						<span className='mc-header__value'>{sessionId}</span>
-					</div>
-					<div className='mc-header__item messageId' title={`ID: ${messageId}`}>
-						<span className='mc-header__value'>{messageId}</span>
+	}: MessageCardBaseProps) => {
+		const { messageId, timestamp, messageType, sessionId, direction, bodyBase64, body } = message;
+
+		const splittedMessageId = messageId.split(':');
+		const messageIdWithHighlightedSession = splittedMessageId.map(
+			getMessageIdWithHighlightedSession,
+		);
+
+		const rootClass = createBemBlock(
+			'message-card-wrapper',
+			isAttached ? 'attached' : null,
+			isBookmarked ? 'pinned' : null,
+			isHighlighted ? 'highlighted' : null,
+			isSoftFiltered ? 'soft-filtered' : null,
+		);
+
+		// session arrow color, we calculating it for each session from-to pair, based on hash
+		const sessionArrowStyle: React.CSSProperties = {
+			display: 'inline-flex',
+			filter: `invert(1) sepia(1) saturate(5) hue-rotate(${calculateHueValue(sessionId)}deg)`,
+		};
+
+		const sessionClass = createStyleSelector(
+			'mc-header__icon mc-header__direction-icon',
+			direction?.toLowerCase(),
+		);
+
+		const iconsClassName = createBemBlock(
+			'message-card-icons',
+			isAttached ? 'attached' : null,
+			isBookmarked ? 'pinned' : null,
+		);
+
+		const renderInlineMessageInfo = () => {
+			if (viewType === MessageViewType.ASCII || viewType === MessageViewType.JSON) {
+				const formattedTimestamp = formatTime(timestampToNumber(timestamp));
+				return (
+					<>
+						{isExport ? (
+							<Checkbox
+								id={messageId}
```

Comment:
There should be stopPropagation() otherwise `addMessageToExport` will be called twice, in Checkbox component and in root div of MessageCardBase